src/dials/command_line/combine_experiments.py

```diff
@@ -752,9 +752,9 @@ def save_in_batches(
                 threshold=params.clustering.threshold,
             )
             n_clusters = len(clustered)
-            clusters = sorted(clustered.clusters, key=len, reverse=True)[
-                : min(params.clustering.max_clusters, n_clusters)
-            ]
+            max_c = params.clustering.max_clusters
+            to_c = min(max_c, n_clusters) if max_c is not None else n_clusters
+            clusters = sorted(clustered.clusters, key=len, reverse=True)[:to_c]
```

Comment:
Absolutely, this looks so much better! I focused on solving the issue at hand, but in hindsight my solution is not very readable. However, I'd suggest strictly checking for `None`. Although it might make little sense for a human to `clustering.use=True` with `clustering.max_clusters=0`, I'd expect `clustering.max_clusters=0` not to produce any clustered data instead of acting as `None`. Might be easier way to effectively achieve `clustering.use=False` in some potential automated pipeline.